Update Button Overview (#646)
telerik/blazor-docs: components/button/overview.md

# Button Overview

> This article introduces the <a href="https://www.telerik.com/blazor-ui/buttons" target="_blank">Blazor Button component</a>, demonstrates how to start using it in a project, and lists its core features.

Reviewer: Simplify

```suggestion
This article introduces the <a href="https://www.telerik.com/blazor-ui/buttons" target="_blank">Blazor Button component</a>, shows how to start using it, and lists its core features.
```

> Buttons communicate specific actions that the user can take. You can set them to display arbitrary content.

Reviewer: This sounds like a too complex and academic explanation for such a simple component.

```suggestion
Buttons convey user actions and can display various content.
```

The Blazor Button provides a variety of styling options through the [built-in themes]({%slug general-information/themes%}) and the button [type]({%slug button-type%}). It supports different types of [icons]({%slug button-icons%}) and generates click [events]({%slug button-events%}).

## Creating Blazor Button

1. Use the `<TelerikButton>` tag to add the component to your razor page.

> 1. Configure the `OnClick` event handling to show the current date and time.

Reviewer: ```suggestion
1. Add an `OnClick` event handler to show the current date and time.
```

> >caption Basic Blazor Button with OnClick event handling

Reviewer: ```suggestion
>caption Basic Blazor Button with OnClick event handler
```

}
````

> To see the result from the code snippet above, select the **PREVIEW** tab in the code snippet toolbar.

Reviewer: Do we need to educate our readers for this? The UI should be user-friendly by itself.

## Icons

> To visually communicate the purpose of a button, you can use add an image, a sprite, or font icon. You can choose between the wide range of built-in font icons or even use your custom font icons. [Read more about the Blazor Button icons...]({%slug button-icons%})

Reviewer: ```suggestion
To visually communicate the purpose of a button, you can add an image, sprite, or font icon. You can choose between a wide range of built-in font icons or use your custom font icons. [Read more about the Blazor Button icons...]({%slug button-icons%})
```

## Type

> To control the desired behavior of the Blazor Button, you can use the `type` attribute. This allows you to select on of the following button types: `Submit`, `Reset`, and `Button`. [Read more about the Blazor Button type...]({%slug button-type%})

Reviewer: ```suggestion
To control the submit behavior of the Blazor Button, use the `Type` attribute. Select from the following button types: `Submit`, `Reset`, and `Button`. [Read more about the Blazor Button Type...]({%slug button-type%})
```

## Events

The Blazor Button generate events that you can handle and further customize the behavior of the Button. [Read more about the Blazor Button events...]({%slug button-events%}).

## Disabled State

> To stop the users from interacting with a button until certain requirements are met, you can render a disabled button. [Read more about the disabled Blazor Button...]({%slug disabled-button%}).

Reviewer: ```suggestion
To stop the users from interacting with a button until certain requirements are met, you can render a disabled button. [Read more about the disabled Blazor Button...]({%slug button-disabled%}).
```

> To stop the users from interacting with a button until certain requirements are met, you can render a disabled button. [Read more about the disabled Blazor Button...]({%slug button-disabled%}).

Reviewer: Simplify

```suggestion
To prevent user interaction with a Button, disable it. [Read more about the disabled Blazor Button...]({%slug button-disabled%}).
```

## Styling

> To customize the style and the appearance of the Blazor Button, you can use the [built-in themes]({%slug general-information/themes%}). Additionally, you can also take advantage of the `Class` attribute and apply custom CSS rules. [Read more about the Blazor Button styling...]({%slug button-styling%})

Reviewer: Themes are not designed to customize the appearance of a single component, so I wonder if we should advertise this option at all, or at least not as a first option.

```suggestion
To customize the style and the appearance of the Blazor Button, you can use the [built-in themes]({%slug general-information/themes%}). Additionally, set the `Class` attribute and implement custom CSS rules. [Read more about the Blazor Button styling...]({%slug button-styling%})
```

## Using Blazor Button Reference

> You can create a reference to an instance of the Blazor Button and use its methods. To declare a reference to the Button component, you must provide the component's namespace. 

Reviewer: Shorter and imperative. The namespace should already be present in _Imports.razor (as per our First Steps), so there is no need for "using" or fully qualified namespaces in the `@code` section. We should consider a general simplification here.

```suggestion
Add a reference to the Button instance to use its methods (for example - `FocusAsync()`).
```

>caption Component namespace and reference

````CSHTML
@using Telerik.Blazor.Components

> <TelerikButton @ref="theButton">Hello!</TelerikButton>
> 
> @code{
> 	Telerik.Blazor.Components.TelerikButton theButton;
> }

Reviewer: ```suggestion
<TelerikButton @ref="theButton">Hello!</TelerikButton>

@code{
    Telerik.Blazor.Components.TelerikButton theButton { get; set; }

    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        await theButton.FocusAsync();

        await base.OnAfterRenderAsync(firstRender);
    }
}
```

````

## Next Steps

* [Styling the Blazor Button]({%slug button-styling%})

* [Using Button Icons]({%slug button-icons%})

## See Also

  * [Live Demo: Button](https://demos.telerik.com/blazor-ui/button/index)
  * [Events]({%slug button-events%})
  * [Type]({%slug button-type%})

>   * [API Reference](https://docs.telerik.com/blazor-ui/api/Telerik.Blazor.Components.TelerikButton)   

Reviewer: ```suggestion
  * [API Reference](https://docs.telerik.com/blazor-ui/api/Telerik.Blazor.Components.TelerikButton)
```
